### PR TITLE
CFE-2826: Trigger promises.cf as the last step of bootstrap

### PR DESCRIPTION
**Pull request by vpodzime**
So that more complete data is available for the first report.

### CODE REVIEW

File: libpromises/failsafe.cf [hunk 1]
```diff
@@ -69,6 +69,13 @@ bundle agent main
                     happen with the most up to date and complete
                     information.";
 
+      "Trigger Policy"
+        usebundle => failsafe_cfe_internal_trigger_policy,
```
Comment (nickanderson) [inline libpromises/failsafe.cf]: So you think it will be best to do this from embedded policy than to have it done from the C?

Comment (vpodzime) [reply libpromises/failsafe.cf]: I don't know yet. As I said, this is the first shot. But I think it makes sense to run `update.cf` and `promises.cf` in the same way.

**Comment by vpodzime** [comment]
Seems to be working nicely. The only thing missing that I'm aware of is tests to make sure that `promises.cf` really is triggered by default during bootstrap.

**Comment by vpodzime** [comment]
> @vpodzime Have you been able to see a difference in MP or similar?

Not really. But I don't quite know how to trigger the worst-case scenario. Things will/would be a lot clearer with MP gathering the data sooner.

**Comment by vpodzime** [comment]
Added check to the deployment tests: https://github.com/cfengine/system-testing/pull/106

**Comment by vpodzime** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/927/)

File: cf-agent/cf-agent.c [hunk 2]
```diff
@@ -587,6 +589,10 @@ static GenericAgentConfig *CheckOpts(int argc, char **argv)
                 }
                 config->agent_specific.agent.show_evaluated_variables = xstrdup(optarg);
             }
+            else if (strcmp(OPTIONS[longopt_idx].name, "skip-bootstrap-policy-run") == 0)
```
Comment (olehermanse) [inline cf-agent/cf-agent.c]: `StringSafeEqual()`

File: libpromises/generic_agent.h [hunk 3]
```diff
@@ -87,6 +87,7 @@ typedef struct
             char *bootstrap_port;
             char *bootstrap_ip;
             bool bootstrap_trust_server;
+            bool bootstrap_trigger_policy;
```
Comment (olehermanse) [inline libpromises/generic_agent.h]: maybe invert this as well, so it's all skip, like the option? I dunno.

File: cf-agent/cf-agent.c (hunk 2 above)
Comment (vpodzime) [reply cf-agent/cf-agent.c]: Yeah, but I prefer the similarity with the surrounding checks in this case.

File: libpromises/generic_agent.h (hunk 3 above)
Comment (vpodzime) [reply libpromises/generic_agent.h]: I was thinking the same, but then I saw we had a "constructor" that sets the defaults. So there's no technical reason to have it inverted and I think it is better readable like this.

**Comment by vpodzime** [comment]
> I'm a little concerned that you haven't seen any difference

I am willing to believe it was just due to the MP cache.

**Pull request merged**